### ISSUE TITLE
find and rm only the relevant files when a test fails

### ISSUE THREAD

**Issue by joshrabinowitz** [sobolevn/git-secret#360]
As mentioned in #334 , when tests fail users might see a long string of find errors, such as:

````
not ok 28 run 'hide' from inside subdirectory
# (in test file /home/joshr/gitsrc/git-secret/tests/test_hide.bats, line 88)
#   `[ "$status" -eq 0 ]' failed
# rm: cannot remove ‘space file’: No such file or directory
# fatal: Not a git repository: '.git'
# find: ‘/tmp/tmp.aBlVDAbUQF’: Permission denied
# find: ‘/tmp/systemd-private-8b7dddc415874f87ad46f53ff94b4d7f-rtkit-daemon.service-KNfvrc’: Permission denied
# find: ‘/tmp/systemd-private-8b7dddc415874f87ad46f53ff94b4d7f-colord.service-ivZa8u’: Permission denied
# find: ‘/tmp/systemd-private-8b7dddc415874f87ad46f53ff94b4d7f-fwupd.service-07aFP3’: Permission denied
# find: ‘/tmp/tmp.UbZvBTtYAa’: Permission denied
# find: ‘/tmp/systemd-private-8b7dddc415874f87ad46f53ff94b4d7f-chronyd.service-G1ov1r’: Permission denied
# find: ‘/tmp/systemd-private-8b7dddc415874f87ad46f53ff94b4d7f-cups.service-kAx3OS’: Permission denied
# find: ‘/tmp/systemd-private-8b7dddc415874f87ad46f53ff94b4d7f-bolt.service-IZaN9N’: Permission denied
# find: ‘/tmp/tmp.1L9d7x7l3I’: Permission denied
````

This shouldn't happen. See also #333

**Comment by Mohjive** [comment]
It would be easy to add a `2>/dev/null` at the end of the `find | xargs rm` command in **function _find_and_clean** at line ~442 in src/_utils/_git_secret_tools.sh to hide all such warnings. The verbose output would still print. Is this an OK solution?

**Comment by joshrabinowitz** [comment]
No let's not do that. 

If a test fails, we should not trying to 'rm' files in /tmp that had nothing to do with our code

**Comment by joshrabinowitz** [comment]
See Also: #452

**Comment by joshrabinowitz** [comment]
This is fixed in #457.

**Comment by joshrabinowitz** [comment]
Fixed by #473
